clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AsgLifecycleHookWorker.groovy

```diff
@@ -44,26 +46,42 @@ class AsgLifecycleHookWorker {
       return
     }
 
+    def targetAccountId = targetCredentials.accountId
     AmazonAutoScaling autoScaling = amazonClientProvider.getAutoScaling(targetCredentials, targetRegion, true)
     lifecycleHooks.each { lifecycleHook ->
       String lifecycleHookName = lifecycleHook.name ?: [targetAsgName, 'lifecycle', idGenerator.nextId()].join('-')
-      def request = new PutLifecycleHookRequest(
-        autoScalingGroupName: targetAsgName,
-        lifecycleHookName: lifecycleHookName,
-        roleARN: lifecycleHook.roleARN,
-        notificationTargetARN: arnRegionTemplater(lifecycleHook.notificationTargetARN, targetRegion),
-        notificationMetadata: lifecycleHook.notificationMetadata,
-        lifecycleTransition: lifecycleHook.lifecycleTransition.toString(),
-        heartbeatTimeout: lifecycleHook.heartbeatTimeout,
-        defaultResult: lifecycleHook.defaultResult.toString()
-      )
-      autoScaling.putLifecycleHook(request)
 
-      task.updateStatus "AWS_DEPLOY", "Creating lifecycle hook (${request}) on ${targetRegion}/${targetAsgName}"
+      switch (lifecycleHook.lifecycleTransition.type) {
+        case AmazonAsgLifecycleHook.TransitionType.LIFECYCLE:
+          def request = new PutLifecycleHookRequest(
+            autoScalingGroupName: targetAsgName,
+            lifecycleHookName: lifecycleHookName,
+            roleARN: arnTemplater(lifecycleHook.roleARN, targetRegion, targetAccountId),
+            notificationTargetARN: arnTemplater(lifecycleHook.notificationTargetARN, targetRegion, targetAccountId),
+            notificationMetadata: lifecycleHook.notificationMetadata,
+            lifecycleTransition: lifecycleHook.lifecycleTransition.toString(),
+            heartbeatTimeout: lifecycleHook.heartbeatTimeout,
+            defaultResult: lifecycleHook.defaultResult.toString()
+          )
+          autoScaling.putLifecycleHook(request)
+
+          task.updateStatus "AWS_DEPLOY", "Creating lifecycle hook (${request}) on ${targetRegion}/${targetAsgName}"
+          break
+
+        case AmazonAsgLifecycleHook.TransitionType.NOTIFICATION:
+          def request = new PutNotificationConfigurationRequest()
+            .withAutoScalingGroupName(targetAsgName)
+            .withNotificationTypes(lifecycleHook.lifecycleTransition.toString())
+            .withTopicARN(arnTemplater(lifecycleHook.notificationTargetARN, targetRegion, targetAccountId))
+          autoScaling.putNotificationConfiguration(request)
+
+          task.updateStatus "AWS_DEPLOY", "Creating notification hook (${request}) on ${targetRegion}/${targetAsgName}"
+          break
+      }
     }
   }
 
-  private static String arnRegionTemplater(String arnTemplate, String region) {
-    arnTemplate.replaceAll(REGION_TEMPLATE_PATTERN, region)
+  private static String arnTemplater(String arnTemplate, String region, String accountId) {
```

Comment:
Turns out I didn't actually need to template `accountId` but figured it didn't hurt to leave in?